### DISCUSSION TITLE
zoau_io module's write() method and writing data to z/OS datasets question

### DISCUSSION THREAD

**Discussion by pawirth** [IBM/zoau-samples#71] [Q&A]
I have just started using Python on z/OS, and have been working with the zoau v1.2.6.0 provided samples. 

I have been reviewing zoau's datasets and zoau_io modules. I have successfully worked with  z/OS datasets using the datasets.read() and datasets.write() methods.  I have also successfully used the zoau_io read method to read a z/OS dataset. The zopen class states write is an option. However, I can't seem to figure out the zoau_io write() method. I have received messages stating that "the object has no attribute 'write'.  

I don't see a write() method in the zoau_io documentation, and I'm just wondering if I have missed something, or if it is not provided yet? The datasets module  seems to have what I need though. 

Thanks!

**Comment by EnsonoStevenPerva** [comment]
As I understand, `zoau_io` is currently read only functionality and write is unavailable. I hope there are plans for write in the future!

**Reply by AnthonyGiorgio** [reply]
There's [an RFE for this idea](https://ibm-z-software-portal.ideas.ibm.com/ideas/ZOAU-I-22) if you'd like to vote for it.

**Comment by pawirth** [comment]
Thanks for the fast response!

I was also looking for a feature to handle z/OS packed decimal data, but did not see anything for that either. From my reading about zoau, it seems we are still at the "getting started mode" with zoau 1.2. I have been using z/OS Java and the JZOS feature, it can import COBOL copybooks into Java and I think I'm going to continue to use that.

**Reply by pawirth** [reply]
I forgot to state that I found a package called pyzfile for z/OS datasets too. It seems that pyzfile is a wrapper for the z/OS C/C++ code.
